File: frontend/src/api/network.ts
import { api } from './client'

export interface NetworkTestRequest {
  host: string
  port: number
  timeout: number
}

export interface NetworkTestResult {
  success: boolean
  host: string
  port: number
  latency_ms: number | null
  error: string | null
}

export function testNetwork(data: NetworkTestRequest) {
  return api.post<NetworkTestResult>('/network/test', data)
}

File: backend/app/schemas/network.py
from pydantic import BaseModel, Field


class NetworkTestRequest(BaseModel):
    host: str
    port: int = Field(default=80, ge=1, le=65535)
    timeout: float = Field(default=5.0, ge=0.5, le=30.0)


class NetworkTestResponse(BaseModel):
    success: bool
    host: str
    port: int
    latency_ms: float | None = None
    error: str | None = None
